File: objective-c/StoreKit/StoreKit_v2/StoreKit_v2/ViewControllers/AMainViewController.m
//
//  AMainViewController.m
//  StoreKit_v2
//
//  Created by Alexandr Kirilov on 15/12/2022.
//

#import "AMainViewController.h"

@implementation AMainViewController

-(void) loadView {
    
    [super loadView];
    [self.view setBackgroundColor:__COLOR_ORANGE];
    [self.pViewContent setBackgroundColor:__COLOR_FIOLENT];
    [self setTitle:@"StoreKit_v2"];
}

@end

File: swift/AsyncAwait/AsyncAwait_v1/AsyncAwait_v1/ViewControllers/AMainViewController.swift
//
//  AMainViewController.swift
//  AsyncAwait_v1
//
//  Created by Alexandr Kirilov on 20/12/2022.
//

import UIKit



class AMainViewController: AViewControllerTemplate {
	
	let pAsyncAwaitView: AAsyncAwait = AAsyncAwait();
	
	override func loadView() {
		
		super.loadView();
		self.title = "AsyncAwait v1";
	}
	
	override func viewDidLoad() {
		
		super.viewDidLoad();
		
		pViewContent.addSubview(pAsyncAwaitView);
		pAsyncAwaitView.topAnchor.constraint(equalTo: pViewContent.topAnchor).isActive = true;
		pAsyncAwaitView.bottomAnchor.constraint(equalTo: pViewContent.bottomAnchor).isActive = true;
		pAsyncAwaitView.leftAnchor.constraint(equalTo: pViewContent.leftAnchor).isActive = true;
		pAsyncAwaitView.rightAnchor.constraint(equalTo: pViewContent.rightAnchor).isActive = true;
		
		Task { [weak self] in
			let oUsers = await pAsyncAwaitView.mFetchUsers();
			DispatchQueue.main.async {
				self?.pAsyncAwaitView.mReloadData(inData: oUsers);
			}
		}
	}
}
